From 7f41f78530577fbcbbf68277a3f8dff770896a74 Mon Sep 17 00:00:00 2001
From: Andrey Kasaurov <andrey.kasaurov@amd.com>
Date: Sat, 25 Jul 2026 01:34:58 -0400
Subject: [PATCH] Use native CMake HIP language for c10d_hip_test

---
 test/cpp/c10d/CMakeLists.txt | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/test/cpp/c10d/CMakeLists.txt b/test/cpp/c10d/CMakeLists.txt
index 24ab667..39b12b2 100644
--- a/test/cpp/c10d/CMakeLists.txt
+++ b/test/cpp/c10d/CMakeLists.txt
@@ -4,8 +4,11 @@ if(USE_CUDA)
   target_link_libraries(c10d_cuda_test torch_cuda)
   add_dependencies(c10d_cuda_test torch_cuda)
 elseif(USE_ROCM)
-  set_source_files_properties(hip/HIPTest.hip PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
-  hip_add_library(c10d_hip_test hip/HIPTest.hip)
+  set_source_files_properties(hip/HIPTest.hip PROPERTIES LANGUAGE HIP)
+  add_library(c10d_hip_test hip/HIPTest.hip)
+  set_target_properties(c10d_hip_test PROPERTIES
+    LINKER_LANGUAGE CXX
+    HIP_ARCHITECTURES "${PYTORCH_ROCM_ARCH}")
   target_include_directories(c10d_hip_test PRIVATE $<BUILD_INTERFACE:${TORCH_SRC_DIR}/csrc/distributed>)
   target_link_libraries(c10d_hip_test torch_hip)
   add_dependencies(c10d_hip_test torch_hip)
-- 
2.34.1

