From b867c62de847d43dae432d6365123b52bf6c3ac1 Mon Sep 17 00:00:00 2001
From: root <root@r18.amd.com>
Date: Sat, 16 May 2026 22:31:55 +0000
Subject: [PATCH] kfdtest: add llvm API ifdef for createMCAsmParser

---
 .../libhsakmt/tests/kfdtest/src/Assemble.cpp             | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/projects/rocr-runtime/libhsakmt/tests/kfdtest/src/Assemble.cpp b/projects/rocr-runtime/libhsakmt/tests/kfdtest/src/Assemble.cpp
index 1f62a8ea9f..0b018e7ffc 100644
--- a/projects/rocr-runtime/libhsakmt/tests/kfdtest/src/Assemble.cpp
+++ b/projects/rocr-runtime/libhsakmt/tests/kfdtest/src/Assemble.cpp
@@ -334,7 +334,11 @@ int Assembler::RunAssemble(const char* const AssemblySource) {
 
     // Set up the MCContext for creating symbols and MCExpr's
 #if LLVM_VERSION_MAJOR > 12
+#if LLVM_MAIN_REVISION >= 577912 // commit # when createMCAsmParser API changed
+    MCContext Ctx(TheTriple, *MAI, *MRI, *STI, &SrcMgr, &MCOptions);
+#else
     MCContext Ctx(TheTriple, MAI.get(), MRI.get(), STI.get(), &SrcMgr, &MCOptions);
+#endif
 #else
     MCObjectFileInfo MOFI;
     MCContext Ctx(MAI.get(), MRI.get(), &MOFI, &SrcMgr, &MCOptions);
@@ -376,8 +380,13 @@ int Assembler::RunAssemble(const char* const AssemblySource) {
             createMCAsmParser(SrcMgr, Ctx, *Streamer, *MAI));
 
     // Set parser to target parser and run
+#if LLVM_MAIN_REVISION >= 577912 // commit # when createMCAsmParser API changed
+    std::unique_ptr<MCTargetAsmParser> TAP(
+            TheTarget->createMCAsmParser(*STI, *Parser, *MCII));
+#else
     std::unique_ptr<MCTargetAsmParser> TAP(
             TheTarget->createMCAsmParser(*STI, *Parser, *MCII, MCOptions));
+#endif
     if (!TAP) {
         outs() << "ASM Error: no assembly parsing support for target " << MCPU << "\n";
         return -1;
-- 
2.49.0

