From 971f69099c2576a66954d9ef2c233bed22d76351 Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Fri, 3 Apr 2026 01:43:51 +0000
Subject: [PATCH] clr ppc64le

---
 .../opencl/tests/ocltst/module/perf/OCLPerfKernelThroughput.h  | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/projects/clr/opencl/tests/ocltst/module/perf/OCLPerfKernelThroughput.h b/projects/clr/opencl/tests/ocltst/module/perf/OCLPerfKernelThroughput.h
index b23ada56fb..a5d3939786 100644
--- a/projects/clr/opencl/tests/ocltst/module/perf/OCLPerfKernelThroughput.h
+++ b/projects/clr/opencl/tests/ocltst/module/perf/OCLPerfKernelThroughput.h
@@ -31,7 +31,10 @@
 #define UNSIGNED_LARGE_INT unsigned long long
 #define MAX_LOOP_ITER 10
 typedef cl_float4 float4;
+
+#if defined(__x86_64__)
 typedef void (*CPUKernel)(__m128*, __m128*, unsigned int);
+#endif
 
 class OCLPerfKernelThroughput : public OCLTestImp {
  public:
-- 
2.47.3

