From e997f025a4cc1f164fe13287e223d4665d60c851 Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Tue, 12 May 2026 19:55:34 +0000
Subject: [PATCH] hip ppc64le

---
 projects/hip/VERSION | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/projects/hip/VERSION b/projects/hip/VERSION
index b094b7acac..6552f37f0b 100644
--- a/projects/hip/VERSION
+++ b/projects/hip/VERSION
@@ -3,4 +3,4 @@
 #HIP_VERSION_MINOR
 13
 #HIP_VERSION_PATCH
-0
+60490
-- 
2.47.3

