From 2c13a769f5cadaf918160800106ca2f92f8f46c0 Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Fri, 29 May 2026 20:05:08 +0000
Subject: [PATCH] rccl ppc64le

---
 projects/rccl/src/init.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/projects/rccl/src/init.cc b/projects/rccl/src/init.cc
index fd84fcb788..20cb2297c7 100644
--- a/projects/rccl/src/init.cc
+++ b/projects/rccl/src/init.cc
@@ -75,7 +75,9 @@
 #include "latency_profiler/CollTraceFunc.h"
 #include "dda_all_reduce_ipc.h"
 #include "ipc_init.h"
+#if defined(__x86_64__) || defined(__i386__)
 #include  <cpuid.h>
+#endif
 
 #ifndef STR2
   #define STR2(v) #v
@@ -255,6 +257,7 @@ static ncclResult_t ncclInit() {
     }
     INFO(NCCL_INIT, "Kernel version: %s", verStr);
     if (strstr(verStr, "cray") == NULL) {
+#if defined(__x86_64__) || defined(__i386__)
       unsigned int eax, ebx, ecx, edx;
       if (!__get_cpuid(1, &eax, &ebx, &ecx, &edx))
         ecx = 0; // cpuid not supported
@@ -272,6 +275,7 @@ static ncclResult_t ncclInit() {
         if (strstr(strValue, "iommu=pt") == NULL)
           WARN("Missing \"iommu=pt\" from kernel command line which can lead to system instablity or hang!");
       }
+#endif
 #ifndef HIP_UNCACHED_MEMORY
       char *env = getenv("HSA_FORCE_FINE_GRAIN_PCIE");
       if (env == NULL || strcmp(env, "1") != 0)
-- 
2.52.0

