From 9878af57bcb88ff1de11f1c74d1b9e06932338bd Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Thu, 28 May 2026 19:23:55 +0000
Subject: [PATCH] Guard x86-only code in rccl

---
 projects/rccl/src/init.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/projects/rccl/src/init.cc b/projects/rccl/src/init.cc
index 29470d0480..5ec487c5fc 100644
--- a/projects/rccl/src/init.cc
+++ b/projects/rccl/src/init.cc
@@ -72,7 +72,9 @@
 
 #include "latency_profiler/CollTrace.h"
 #include "latency_profiler/CollTraceFunc.h"
+#if defined(__x86_64__) || defined(__i386__)
 #include  <cpuid.h>
+#endif
 
 #ifndef STR2
   #define STR2(v) #v
@@ -251,6 +253,7 @@ static ncclResult_t ncclInit() {
     }
     INFO(NCCL_INIT, "Kernel version: %s", verStr);
     if (strstr(verStr, "cray") == NULL) {
+#if defined(__x86_64__) || defined(__i386__)
       unsigned int eax, ebx, ecx, edx;
       if (!__get_cpuid(1, &eax, &ebx, &ecx, &edx))
         ecx = 0; // cpuid not supported
@@ -268,6 +271,7 @@ static ncclResult_t ncclInit() {
         if (strstr(strValue, "iommu=pt") == NULL)
           WARN("Missing \"iommu=pt\" from kernel command line which can lead to system instablity or hang!");
       }
+#endif
 #ifndef HIP_UNCACHED_MEMORY
       char *env = getenv("HSA_FORCE_FINE_GRAIN_PCIE");
       if (env == NULL || strcmp(env, "1") != 0)
-- 
2.52.0

