From 49c4ae8d53bc05905702b8e015fcbd59bf89eef4 Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Thu, 2 Apr 2026 21:15:16 +0000
Subject: [PATCH] rocprofiler-compute ppc64le

---
 projects/rocprofiler-compute/CMakeLists.txt | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/projects/rocprofiler-compute/CMakeLists.txt b/projects/rocprofiler-compute/CMakeLists.txt
index d6a0ad7fcc..160cdf1c63 100644
--- a/projects/rocprofiler-compute/CMakeLists.txt
+++ b/projects/rocprofiler-compute/CMakeLists.txt
@@ -198,13 +198,12 @@ if(LOCALHOST MATCHES "TheraS01|.*\.thera\.amd\.com|thera-hn")
 endif()
 
 # Setup testing collateral
-
 option(ENABLE_TESTS "Enable compilation of testing collateral" OFF)
 set(CMAKE_HIP_FLAGS_RELEASE "-O2")
-if(${ENABLE_TESTS})
-    enable_language("C" "HIP")
-    add_subdirectory(tests)
-endif()
+#if(${ENABLE_TESTS})
+#    enable_language("C" "HIP")
+#    add_subdirectory(tests)
+#endif()
 message(STATUS "Enable tests compilation: ${ENABLE_TESTS}")
 
 enable_testing()
-- 
2.47.3

