From 94c5ba7dd0c9f5f7065effb0b6150b3fd7901d5d Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Tue, 31 Mar 2026 23:24:05 +0000
Subject: [PATCH] ESMI x86_64 only

---
 projects/amdsmi/CMakeLists.txt | 19 +++++++------------
 1 file changed, 7 insertions(+), 12 deletions(-)

diff --git a/projects/amdsmi/CMakeLists.txt b/projects/amdsmi/CMakeLists.txt
index 081bcf970f..20ce609ca8 100644
--- a/projects/amdsmi/CMakeLists.txt
+++ b/projects/amdsmi/CMakeLists.txt
@@ -115,7 +115,13 @@ include(GNUInstallDirs)
 
 option(BUILD_TESTS "Build test suite" OFF)
 option(ENABLE_ASAN_PACKAGING "" OFF)
-option(ENABLE_ESMI_LIB "Build ESMI Library" ON)
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
+    option(ENABLE_ESMI_LIB "Build ESMI Library" ON)
+else()
+    option(ENABLE_ESMI_LIB "Build ESMI Library" OFF)
+endif()
+
 option(BUILD_EXAMPLES "Build examples" OFF)
 
 # If amdsmi is built as a static library, it should support being embedded in other programs. The setting below essentially enables the -fPIC flag.
@@ -287,17 +293,6 @@ if(ENABLE_ESMI_LIB)
         endif()
     endif()
 
-    if(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
-        execute_process(
-            COMMAND ${GIT_EXECUTABLE} apply ${CMAKE_SOURCE_DIR}/../../../patches/esmi/esmi_ib_library_e_smi.c.patch
-            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/esmi_ib_library
-            RESULT_VARIABLE PATCH_RESULT
-            OUTPUT_VARIABLE PATCH_OUTPUT
-            ERROR_VARIABLE PATCH_ERROR
-            COMMAND_ERROR_IS_FATAL ANY
-        )
-    endif()
-
     # Make sure to update the amd_hsmp.h file with the corresponding esmi version
     file(
         COPY "${PROJECT_SOURCE_DIR}/include/amd_smi/impl/amd_hsmp.h"
-- 
2.47.3

