From 45429d512e4ee2a650c267c9d082fddea3122af3 Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Thu, 2 Apr 2026 19:00:05 +0000
Subject: [PATCH] Remove bf16 workaround for ppc64le

---
 projects/hip/include/hip/hip_bf16.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/projects/hip/include/hip/hip_bf16.h b/projects/hip/include/hip/hip_bf16.h
index ecc11a5d7b..7e6b827314 100644
--- a/projects/hip/include/hip/hip_bf16.h
+++ b/projects/hip/include/hip/hip_bf16.h
@@ -10,11 +10,7 @@
 #include <hip/hip_common.h>
 
 #if defined(__HIP_PLATFORM_AMD__) && !defined(__HIP_PLATFORM_NVIDIA__)
-#if defined(__powerpc64__) || defined(__PPC64__)
-#include <hip/amd_detail/amd_ppc64le_hip_bf16.h>
-#else
 #include <hip/amd_detail/amd_hip_bf16.h>
-#endif
 #elif !defined(__HIP_PLATFORM_AMD__) && defined(__HIP_PLATFORM_NVIDIA__)
 #include <hip/nvidia_detail/nvidia_hip_bf16.h>
 #else
-- 
2.47.3

