From 0ef888d0702992750403d8627671debbe8357042 Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Fri, 3 Jul 2026 20:17:44 +0000
Subject: [PATCH] hipblas ppc64le

---
 projects/hipblas/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/projects/hipblas/CMakeLists.txt b/projects/hipblas/CMakeLists.txt
index 8ec3d2259ac..50b68f3f355 100644
--- a/projects/hipblas/CMakeLists.txt
+++ b/projects/hipblas/CMakeLists.txt
@@ -43,6 +43,9 @@ if( NOT DEFINED CMAKE_CONFIGURATION_TYPES AND NOT DEFINED CMAKE_BUILD_TYPE )
 endif()
 
 project( hipblas LANGUAGES CXX )
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
+  add_compile_options("-DNO_WARN_X86_INTRINSICS")
+endif()
 
 find_package(Python3 3.9 COMPONENTS Interpreter REQUIRED)
 
-- 
2.52.0

