From 4ef6ecba61f37e08aacdb2c9bdc0e92681ce3d79 Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Thu, 26 Mar 2026 19:38:27 +0000
Subject: [PATCH] hipfft ppc64le

---
 projects/hipfft/CMakeLists.txt | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/projects/hipfft/CMakeLists.txt b/projects/hipfft/CMakeLists.txt
index 4efdf121ac..f27441acea 100644
--- a/projects/hipfft/CMakeLists.txt
+++ b/projects/hipfft/CMakeLists.txt
@@ -191,6 +191,20 @@ if( BUILD_CLIENTS )
   set( BUILD_CLIENTS_TESTS ON )
 endif()
 
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
+  # The Alias: Map the problematic _Float16 to the supported __fp16
+  add_compile_definitions("$<$<COMPILE_LANGUAGE:CXX>:_Float16=__fp16>")
+
+  # The Driver Flag: Tell Clang it's okay to pass these in registers
+  set(PPC_PFE_FLAGS "-Xclang" "-fallow-half-arguments-and-returns")
+  add_compile_options("$<$<OR:$<COMPILE_LANGUAGE:CXX>,$<COMPILE_LANGUAGE:C>>:${PPC_PFE_FLAGS}>")
+  add_compile_options("$<$<OR:$<COMPILE_LANGUAGE:CXX>,$<COMPILE_LANGUAGE:C>>:-Wno-narrowing>")
+
+  # The Operator Block: Stop the headers from trying to use GPU-only math operators on host
+  add_compile_definitions("$<$<COMPILE_LANGUAGE:CXX>:__HIP_NO_HALF_OPERATORS__=1>")
+  add_compile_definitions("$<$<COMPILE_LANGUAGE:CXX>:__HIP_NO_HALF_CONVERSIONS__=1>")
+endif()
+
 # old name for BUILD_CLIENTS_BENCH
 if( BUILD_CLIENTS_RIDER )
   set( BUILD_CLIENTS_BENCH ${BUILD_CLIENTS_RIDER} )
-- 
2.47.3

