From e8ca3dd4c68c61d8cf9db65611513e41fa5b25ef Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Thu, 26 Mar 2026 19:38:28 +0000
Subject: [PATCH] rocfft ppc64le

---
 projects/rocfft/CMakeLists.txt | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/projects/rocfft/CMakeLists.txt b/projects/rocfft/CMakeLists.txt
index be8a0b71ae..31a277413e 100644
--- a/projects/rocfft/CMakeLists.txt
+++ b/projects/rocfft/CMakeLists.txt
@@ -182,6 +182,22 @@ rocm_check_target_ids(GPU_TARGETS TARGETS "${GPU_TARGETS}")
 find_package( hip REQUIRED CONFIG PATHS /opt/rocm/lib/cmake/hip/ )
 find_package( hiprtc REQUIRED CONFIG PATHS /opt/rocm/lib/cmake/hiprtc/ )
 
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
+  # Add -xhip ONLY to C++ files, ignoring C files like sqlite3.c
+  add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:-xhip>")
+  
+  # The Alias: Map the problematic _Float16 to the supported __fp16
+  add_compile_definitions("$<$<COMPILE_LANGUAGE:CXX>:_Float16=__fp16>")
+
+  # The Driver Flag: Tell Clang it's okay to pass these in registers
+  set(PPC_PFE_FLAGS "-Xclang" "-fallow-half-arguments-and-returns")
+  add_compile_options("$<$<OR:$<COMPILE_LANGUAGE:CXX>,$<COMPILE_LANGUAGE:C>>:${PPC_PFE_FLAGS}>")
+
+  # The Operator Block: Stop the headers from trying to use GPU-only math operators on host
+  add_compile_definitions("$<$<COMPILE_LANGUAGE:CXX>:__HIP_NO_HALF_OPERATORS__=1>")
+  add_compile_definitions("$<$<COMPILE_LANGUAGE:CXX>:__HIP_NO_HALF_CONVERSIONS__=1>")
+endif()
+
 # The nvidia backend can be used to compile for CUDA devices.
 # Specify the CUDA prefix in the CUDA_PREFIX variable.
 # CUDA_ARCH (e.g. sm_75) is also required.
@@ -224,6 +240,12 @@ if( BUILD_CLIENTS )
   set( BUILD_CLIENTS_BENCH ON )
   set( BUILD_CLIENTS_SAMPLES ON )
   set( BUILD_CLIENTS_TESTS ON )
+  else()
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
+    set( BUILD_CLIENTS_BENCH OFF )
+    set( BUILD_CLIENTS_SAMPLES OFF )
+    set( BUILD_CLIENTS_TESTS OFF )
+  endif()
 endif( )
 
 # old name for BUILD_CLIENTS_BENCH
-- 
2.47.3

