From 606cefe3d3b8a5571814907e508470d01a7d38f7 Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Thu, 19 Mar 2026 18:22:11 +0000
Subject: [PATCH] Update e_smi.c to build on ppc64le

---
 src/e_smi.c | 60 +++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 60 insertions(+)

diff --git a/src/e_smi.c b/src/e_smi.c
index d62e8bd..0f1592c 100644
--- a/src/e_smi.c
+++ b/src/e_smi.c
@@ -4,7 +4,9 @@
  * Copyright (c) 2020-2023, Advanced Micro Devices, Inc. All rights reserved.
  *
  */
+#if defined(__x86_64__) || defined(_M_X64)
 #include <cpuid.h>
+#endif
 #include <dirent.h>
 #include <errno.h>
 #include <math.h>
@@ -330,6 +332,7 @@ static esmi_status_t create_cpu_mappings(struct system_metrics *psm)
 	return ESMI_SUCCESS;
 }
 
+#if defined(__x86_64__) || defined(_M_X64)
 static esmi_status_t detect_packages(struct system_metrics *psm)
 {
 	uint32_t eax, ebx, ecx,edx;
@@ -379,6 +382,63 @@ static esmi_status_t detect_packages(struct system_metrics *psm)
 	return ESMI_SUCCESS;
 }
 
+#elif defined(__ppc64le__) || defined(__PPC64__)
+static esmi_status_t detect_packages(struct system_metrics *psysm)
+{
+	int ret;
+	FILE *fp;
+	uint32_t pvr;
+	char str[CPU_INFO_LINE_SIZE];
+
+	if (NULL == psysm) {
+		return ESMI_IO_ERROR;
+	}
+
+	fp = fopen(CPU_INFO_PATH, "r");
+	if (!fp) {
+		return ESMI_FILE_ERROR;
+	}
+
+	fclose(fp);
+
+	asm volatile ("mfspr %0, 287" : "=r" (pvr));
+	psysm->cpu_family = pvr;
+	psysm->cpu_model = pvr;
+
+
+	ret = read_index(CPU_COUNT_PATH);
+	if(ret < 0) {
+	    return ESMI_IO_ERROR;
+	}
+	psysm->total_cores = ret + 1;
+
+	fp = popen("lscpu", "r");
+	if (!fp) {
+	    return ESMI_IO_ERROR;
+	}
+
+	int sockets = -1;
+	char line[256];
+	while (fgets(line, sizeof(line), fp)) {
+	    if (strncmp(line, "Socket(s):", 10) == 0) {
+		sockets = atoi(line + 10);
+		break;
+	    }
+	}
+	pclose(fp);
+
+	if (sockets <= 0) {
+	    return ESMI_IO_ERROR;
+	}
+	psysm->total_sockets = sockets;
+
+	return ESMI_SUCCESS;
+
+}
+#else
+#error "Unsupported architecture"
+#endif
+
 static bool check_for_64bit_rapl_reg(struct system_metrics *psysm)
 {
 	bool ret = true;
-- 
2.47.3

