From 938284692d08f493db685e2c77b1a2bf93f58be8 Mon Sep 17 00:00:00 2001
From: Scott Todd <scott.todd0@gmail.com>
Date: Thu, 4 Sep 2025 15:27:05 -0700
Subject: [PATCH] Set HIP_PLATFORM=amd for aotriton source build.

---
 cmake/External/aotriton.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/External/aotriton.cmake b/cmake/External/aotriton.cmake
index 5d915877465..cf2c7834d48 100644
--- a/cmake/External/aotriton.cmake
+++ b/cmake/External/aotriton.cmake
@@ -62,6 +62,7 @@ if(NOT __AOTRITON_INCLUDED)
       -DCMAKE_INSTALL_PREFIX:FILEPATH=${__AOTRITON_INSTALL_DIR}
       CMAKE_ARGS
       -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+      -DHIP_PLATFORM=amd
       -DAOTRITON_GPU_BUILD_TIMEOUT=0
       -DAOTRITON_NO_PYTHON=ON
       -DAOTRITON_NOIMAGE_MODE=${noimage}
-- 
2.47.1.windows.2

